addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js

```diff
@@ -72,23 +78,28 @@ export class AnalyticDistribution extends Component {
         // and thus different applicabilities apply
         // or a model applies that contains unavailable plans
         // This should only execute when these fields have changed, therefore we use the `_field` props.
-        // (consider including the plans in the computed json, python side)
         const valueChanged = JSON.stringify(this.props.value) !== JSON.stringify(nextProps.value);
-        if (this.applicabilityParamsChanged(nextProps) || valueChanged) {
-            await this.fetchAllPlans(nextProps);
-            await this.formatData(nextProps);
+        if (valueChanged) {
+            if (this.props.force_applicability) {
+                this.formatData(nextProps);
+            } else {
+                this.fetchPlansAndFormatData(nextProps);
+            }
+            return;
         }
-    }
 
-    applicabilityParamsChanged(nextProps) {
-        if (this.props.force_applicability) {
-            return false;
-        }
-        if (this.props.account_field && this.props.record.data[this.props.account_field] !== nextProps.record.data[this.props.account_field] ||
-            this.props.product_field && this.props.record.data[this.props.product_field] !== nextProps.record.data[this.props.product_field]) {
-            return true;
+        const currentAccount = this.props.account_field ? this.props.record.data[this.props.account_field] : false;
+        const currentProduct = this.props.product_field ? this.props.record.data[this.props.product_field] : false;
+        if (!shallowEqual(this.lastProduct, currentProduct) || !shallowEqual(this.lastAccount, currentAccount)) {
```

Comment:
If not `force_applicability` and (value changed or account changed or product changed): `fetchAllPlans`
Instead of heaving 3 if doing that

Reply:
improved the willUpdate function